Clean up E0501 explanation (#70690)
rust-lang/rust: src/librustc_error_codes/error_codes/E0501.md

> A mutable variable is being used whereas it has already been captured by a

Reviewer: ```suggestion
A mutable variable is used but it is already captured by a closure. 
```

> closure.

Reviewer: ```suggestion

```

}
```

> This error indicates that a mutable variable is being used while it is still

Reviewer: ```suggestion
This error indicates that a mutable variable is used while it is still
```

captured by a closure. Because the closure has borrowed the variable, it is not

> available for use until the closure goes out of scope.

Reviewer: ```suggestion
available till the closure goes out of scope.
```
